Code first

szKantorowicz opened the pull request.

szKantorowicz committed:
Stworzenie bazowego projektu

szKantorowicz committed:
Dodanie CreateDatabase oraz poprawki przy ID PRIMARY KEY

szKantorowicz committed:
Stworzenie tabelek

szKantorowicz committed:
Dodanie CONTEXT

szKantorowicz committed:
Dodanie przykładowych  danych

PiotrKantorowicz1 commented on src/BProject/code first/Model/Address.cs:
```diff
@@ -0,0 +1,23 @@
+        public string Postcode { get; set; }
+        public string Country { get; set; }
+        public Nullable<int> Level { get; set; }
+        public Nullable<System.DateTime> UpdatedDate { get; set; }
```
```suggestion
        public System.DateTime? UpdatedDate { get; set; }
```

PiotrKantorowicz1 commented on src/BProject/code first/Model/Address.cs:
```diff
@@ -0,0 +1,23 @@
+        public string Country { get; set; }
+        public Nullable<int> Level { get; set; }
+        public Nullable<System.DateTime> UpdatedDate { get; set; }
+        public Nullable<System.DateTime> CreatedDate { get; set; }
```
```suggestion
        public System.DateTime? CreatedDate { get; set; }
```

PiotrKantorowicz1 commented on src/BProject/code first/Model/Customer.cs:
```diff
@@ -0,0 +1,34 @@
+        }
+
+        public int ID { get; set; }
+        public Nullable<int> UserID { get; set; }
```
```suggestion
        public int? UserID { get; set; }
```

PiotrKantorowicz1 commented on src/BProject/code first/Model/Customer.cs:
```diff
@@ -0,0 +1,34 @@
+        public string UserName { get; set; }
+        public string Email { get; set; }
+        public string Phone { get; set; }
+        public Nullable<System.DateTime> UpdatedDate { get; set; }
```
```suggestion
        public System.DateTime UpdatedDate { get; set; }
```

PiotrKantorowicz1 commented on src/BProject/code first/Model/Customer.cs:
```diff
@@ -0,0 +1,34 @@
+        public string Email { get; set; }
+        public string Phone { get; set; }
+        public Nullable<System.DateTime> UpdatedDate { get; set; }
+        public Nullable<System.DateTime> CreatedDate { get; set; }
```
```suggestion
        public System.DateTime? CreatedDate { get; set; }
```

PiotrKantorowicz1 commented on src/BProject/code first/Model/Order.cs:
```diff
@@ -0,0 +1,35 @@
+        public Nullable<int> PaymentTypeID { get; set; }
+        public Nullable<int> Status { get; set; }
+        public Nullable<System.DateTime> UpdatedDate { get; set; }
+        public Nullable<System.DateTime> CreatedDate { get; set; }
```
Wszystkie Nullable zamień na zmienną z ? np. int?, decimal?, bool?

PiotrKantorowicz1 commented on src/BProject/code first/Model/Order.cs:
```diff
@@ -0,0 +1,35 @@
+        public virtual Customer Customer { get; set; }
+        public virtual PaymentType PaymentType { get; set; }
+        public virtual Status Status1 { get; set; }
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
```
```suggestion
```

PiotrKantorowicz1 commented on src/BProject/code first/Model/Order.cs:
```diff
@@ -0,0 +1,35 @@
+        public Nullable<decimal> TotalAmount { get; set; }
+        public Nullable<bool> IsPayed { get; set; }
+        public Nullable<int> PaymentTypeID { get; set; }
+        public Nullable<int> Status { get; set; }
```
```suggestion
        public Nullable<int> StatusID { get; set; }
```

Będzie trzeba w bazie zmienić, później Status na StatusID

PiotrKantorowicz1 commented on src/BProject/code first/Model/Order.cs:
```diff
@@ -0,0 +1,35 @@
+
+        public virtual Customer Customer { get; set; }
+        public virtual PaymentType PaymentType { get; set; }
+        public virtual Status Status1 { get; set; }
```
```suggestion
        public virtual Status Status { get; set; }
```

PiotrKantorowicz1 commented on src/BProject/code first/Model/OrderItem.cs:
```diff
@@ -0,0 +1,24 @@
+        public Nullable<decimal> TotalPrice { get; set; }
+        public Nullable<int> Quantity { get; set; }
+        public Nullable<System.DateTime> UpdatedDate { get; set; }
+        public Nullable<System.DateTime> CreatedDate { get; set; }
```
To samo Nullable zamień na zmienną + ?

PiotrKantorowicz1 commented on src/BProject/code first/Model/Product.cs:
```diff
@@ -0,0 +1,33 @@
+        public Nullable<System.DateTime> UpdatedDate { get; set; }
+        public Nullable<System.DateTime> CreatedDate { get; set; }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
```
```suggestion
```

PiotrKantorowicz1 commented on src/BProject/code first/Model/Product.cs:
```diff
@@ -0,0 +1,33 @@
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
+        public virtual ICollection<OrderItem> OrderItems { get; set; }
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
```
```suggestion
```

PiotrKantorowicz1 commented on src/BProject/code first/Model/Product.cs:
```diff
@@ -0,0 +1,33 @@
+        public Nullable<int> Quantityinstock { get; set; }
+        public Nullable<bool> Avilability { get; set; }
+        public Nullable<System.DateTime> UpdatedDate { get; set; }
+        public Nullable<System.DateTime> CreatedDate { get; set; }
```
Nullable zamień na zmienna + ?

PiotrKantorowicz1 commented on src/BProject/code first/Model/Role.cs:
```diff
@@ -0,0 +1,25 @@
+        public int ID { get; set; }
+        public string Name { get; set; }
+        public Nullable<System.DateTime> UpdatedDate { get; set; }
+        public Nullable<System.DateTime> CreatedDate { get; set; }
```
Nullable = System.DateTime?

PiotrKantorowicz1 commented on src/BProject/code first/Model/User.cs:
```diff
@@ -0,0 +1,26 @@
+        public string Name { get; set; }
+        public string Email { get; set; }
+        public string Password { get; set; }
+        public Nullable<System.DateTime> UpdatedDate { get; set; }
```
Nullable = zmienna + "?"

szKantorowicz committed:
Poprawka kodu

szKantorowicz committed:
Update src/BProject/code first/Model/Address.cs

Co-Authored-By: Piotr Kantorowicz <piotrkantorowicz@gmail.com>

szKantorowicz committed:
Dodanie poprawek do Visuala (Nullable itp.)

szKantorowicz committed:
Merge remote-tracking branch 'orgin/code_first' into Code_first

szKantorowicz committed:
Poprawka z Consol na Console

szKantorowicz committed:
Dodanie Entity Framework do głownego projektu

szKantorowicz committed:
Dodanie pustych klas CONFIGURATION

szKantorowicz committed:
Skończenie AdressConfiguration

szKantorowicz committed:
Skończenie CustomerConfiguration

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/Order.cs:
```diff
@@ -0,0 +1,35 @@
+        public decimal? TotalAmount { get; set; }
+        public bool? IsPayed { get; set; }
+        public int? PaymentTypeID { get; set; }
+        public int? Status { get; set; }
```
```suggestion
        public int? StatusID { get; set; }
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/Order.cs:
```diff
@@ -0,0 +1,35 @@
+
+        public virtual Customer Customer { get; set; }
+        public virtual PaymentType PaymentType { get; set; }
+        public virtual Status Status1 { get; set; }
```
```suggestion
        public virtual Status Status { get; set; }
```

szKantorowicz committed:
Poprawka ConfigurationAdress Configuration Customer oraz skończenie UserConfiguration

szKantorowicz committed:
Skończony OrderConfiguration

szKantorowicz committed:
Skończone PaymentTypeConfiguration

szKantorowicz committed:
Skończony StatusConfiguration

szKantorowicz committed:
Skończony OrderItemConfiguration

szKantorowicz committed:
Skonczony ProductConfiguration

szKantorowicz committed:
Poprawja ProductConfiguration

szKantorowicz committed:
Skończony Category Configuration

szKantorowicz committed:
Dodanie poprawki z kluczem Product_category

Poprawka klucza, bład przy tworzeniu tablic, należy zamienić pomiędzy OrderItem-Product z many to one na one to many

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/Address.cs:
```diff
@@ -0,0 +1,21 @@
+{
+    public class Address : BaseEntityWithTimestamp
+    {
+        public int? CustomerID { get; set; }
```
CustomerID może być zwykłym int and nie int?, ponieważ deklarujemy, że jest required.

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/Customer.cs:
```diff
@@ -0,0 +1,35 @@
+        public virtual ICollection<Order> Orders
+        {
+            get; set;
+        }
```
``` 
public virtual User User { get; set; }
public virtual ICollection<Address> Addresses { get; set; }
public virtual ICollection<Order> Orders  { get; set; }
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/Customer.cs:
```diff
@@ -0,0 +1,35 @@
+        }
+
+        public int ID { get; set; }
+        public int? UserID { get; set; }
```
UserID też może być zwykłym int, bo Customer nie może istnieć bez Usera.

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/Order.cs:
```diff
@@ -0,0 +1,35 @@
+        }
+
+        public int ID { get; set; }
+        public int? CustomerID { get; set; }
```
CustomerID zwykły int zamiast int?

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/Order.cs:
```diff
@@ -0,0 +1,35 @@
+        public int? PaymentTypeID { get; set; }
+        public int? StatusID { get; set; }
+        public System.DateTime? UpdatedDate { get; set; }
+        public System.DateTime? CreatedDate { get; set; }
```
CreatedDate zawsze zwykły DateTime

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/OrderItem.cs:
```diff
@@ -0,0 +1,24 @@
+     public class OrderItem
+    {
+        public int ID { get; set; }
+        public int? OrderID { get; set; }
```
OrderID też zwykły int

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/OrderItem.cs:
```diff
@@ -0,0 +1,24 @@
+    {
+        public int ID { get; set; }
+        public int? OrderID { get; set; }
+        public int? ProductID { get; set; }
```
ProductID też zwykły int

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/OrderItem.cs:
```diff
@@ -0,0 +1,24 @@
+        public decimal? TotalPrice { get; set; }
+        public int? Quantity { get; set; }
+        public System.DateTime? UpdatedDate { get; set; }
+        public System.DateTime? CreatedDate { get; set; }
```
CreatedDate zawsze DateTime

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/Product.cs:
```diff
@@ -0,0 +1,33 @@
+       
+        public virtual ICollection<OrderItem> OrderItems { get; set; }
+      
+        public virtual ICollection<Category> Categories { get; set; }
```
```
public virtual ICollection<OrderItem> OrderItems { get; set; }
public virtual ICollection<Category> Categories { get; set; }
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/Product.cs:
```diff
@@ -0,0 +1,33 @@
+        public System.DateTime? UpdatedDate { get; set; }
+        public System.DateTime? CreatedDate { get; set; }
+
+       
```
```suggestion
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/Product.cs:
```diff
@@ -0,0 +1,33 @@
+        }
+
+        public int ID { get; set; }
+        public int? Category { get; set; }
```
```suggestion
        public int CategoryID { get; set; }
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/Role.cs:
```diff
@@ -0,0 +1,25 @@
+        public int ID { get; set; }
+        public string Name { get; set; }
+        public System.DateTime? UpdatedDate { get; set; }
+        public System.DateTime? CreatedDate { get; set; }
```
CreatedDate zawsze DateTime

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/User.cs:
```diff
@@ -0,0 +1,26 @@
+        public string Email { get; set; }
+        public string Password { get; set; }
+        public System.DateTime? UpdatedDate { get; set; }
+        public System.DateTime? CreatedDate { get; set; }
```
CreatedDate zawsze DateTime

PiotrKantorowicz1 commented on src/BProject/BProject.Core/Model/User.cs:
```diff
@@ -0,0 +1,26 @@
+        public string Password { get; set; }
+        public System.DateTime? UpdatedDate { get; set; }
+        public System.DateTime? CreatedDate { get; set; }
+        public Customer Customer { get; set; }
```
```suggestion
        public virtual Customer Customer { get; set; }
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/EntityTypeConfiguration/CategoryConfiguration.cs:
```diff
@@ -0,0 +1,30 @@
+
+            Property(ca => ca.Description)
+                .HasMaxLength(1000)
+                .IsRequired();
```
```suggestion
                .IsOptional()
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/EntityTypeConfiguration/CustomerConfiguration.cs:
```diff
@@ -0,0 +1,51 @@
+
+            Property(c => c.FirstName)
+                .HasMaxLength(100)
+                .IsRequired();
```
```suggestion
                .IsOptional();
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/EntityTypeConfiguration/CustomerConfiguration.cs:
```diff
@@ -0,0 +1,51 @@
+
+            Property(c => c.LastName)
+                .HasMaxLength(100)
+                .IsRequired();
```
```suggestion
                .IsOptional();
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/EntityTypeConfiguration/CustomerConfiguration.cs:
```diff
@@ -0,0 +1,51 @@
+
+            Property(c => c.Phone)
+                .HasMaxLength(10)
+                .IsRequired();
```
```suggestion
                .IsOptional();
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/EntityTypeConfiguration/OrderConfiguration.cs:
```diff
@@ -0,0 +1,48 @@
+                .IsRequired();
+
+            Property(o => o.IsPayed)
+                 .IsRequired();
```
```suggestion
                .IsOptional();
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/EntityTypeConfiguration/OrderConfiguration.cs:
```diff
@@ -0,0 +1,48 @@
+                 .IsRequired();
+
+            Property(o => o.PaymentTypeID)
+                 .IsRequired();
```
```suggestion
                .IsOptional();
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/EntityTypeConfiguration/OrderConfiguration.cs:
```diff
@@ -0,0 +1,48 @@
+                 .IsRequired();
+
+            Property(o => o.StatusID)
+                 .IsRequired();
```
```suggestion
                .IsOptional();
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/EntityTypeConfiguration/OrderConfiguration.cs:
```diff
@@ -0,0 +1,48 @@
+                .IsRequired();
+
+            Property(o => o.TotalAmount)
+                .IsRequired();
```
```suggestion
                .IsOptional();
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/EntityTypeConfiguration/PaymentTypeConfiguration.cs:
```diff
@@ -0,0 +1,30 @@
+                .IsRequired();
+
+            HasMany(pt => pt.Orders)
+               .WithRequired(o => o.PaymentType)
```
```suggestion
               .WithOptional(o => o.PaymentType)
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/EntityTypeConfiguration/ProductConfiguration.cs:
```diff
@@ -0,0 +1,58 @@
+                    p.MapLeftKey("CategoryID");
+                    p.MapRightKey("ProductID");
+                    p.ToTable("CategoryProduct");
+                });
```
```
HasMany(p => p.Categories)
                .WithMany(ca => ca.Products)
                .Map(p =>
                {
                    p.MapLeftKey("ProductID");
                    p.MapRightKey("CategoryID");
                    p.ToTable("ProductCategory");
                });
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/EntityTypeConfiguration/StatusConfiguration.cs:
```diff
@@ -0,0 +1,32 @@
+                .IsRequired();
+
+            HasMany(s => s.Orders)
+               .WithRequired(o => o.Status)
```
```suggestion
               .WithOptional(o => o.Status)
```

PiotrKantorowicz1 commented on src/BProject/BProject.Core/BProjectContext.cs:
```diff
@@ -0,0 +1,39 @@
+        protected override void OnModelCreating(DbModelBuilder modelBuilder)
+        {
+            modelBuilder.Configurations.Add(new AddressConfiguration());
+            
```
```
modelBuilder.Configurations.Add(new CategoryConfiguration());
modelBuilder.Configurations.Add(new CustomerConfiguration());
...
```

szKantorowicz committed:
Dodanie dziedziczenia po bazach BaseEntity

szKantorowicz committed:
Dodanie reszty poprawek

szKantorowicz committed:
Poprawa "formatowania"

szKantorowicz committed:
Poprawka project ->Project

szKantorowicz committed:
Dodanie poprawek w formatowaniu

szKantorowicz committed:
Dodanie kolejnych poprawek

szKantorowicz committed:
proprawki

szKantorowicz committed:
Migration
